client/state/partner-portal/partner/actions.ts

```diff
@@ -51,17 +52,29 @@ export function fetchPartner( dispatch: ReduxDispatch, getState: () => PartnerPo
 }
 
 export function receivePartner( partner: Partner ): PartnerPortalThunkAction {
-	return ( dispatch: ReduxDispatch ): void => {
+	return ( dispatch: ReduxDispatch, getState: () => PartnerPortalStore ): void => {
 		dispatch( {
 			type: JETPACK_PARTNER_PORTAL_PARTNER_RECEIVE,
 			partner,
 		} );
 
 		// If we only get a single key, auto-select it for the user for simplicity.
+		// We check the active key otherwise.
 		const keys = partner.keys.map( ( key ) => key.id );
+		const numberOfKeys = keys.length;
+		const currentPartnerKeyId = getActivePartnerKeyId( getState() );
+		// If the current stored partner key id is disabled
+		const isPartnerKeyAvailable = keys.includes( currentPartnerKeyId );
+
+		if ( ! isPartnerKeyAvailable ) {
+			// do something else
+			return;
+		}
 
-		if ( keys.length === 1 ) {
+		if ( numberOfKeys === 1 ) {
 			dispatch( setActivePartnerKey( keys[ 0 ] ) );
+		} else if ( numberOfKeys > 1 ) {
+			dispatch( setActivePartnerKey( currentPartnerKeyId ) );
```

Comment:
I think replacing the logic with this can make it easier to understand what is happening:
```js
const activePartnerKeyId = getActivePartnerKeyId( getState() );
const keys = partner.keys.map( ( key ) => key.id );
let newKeyId = 0;

if ( keys.length === 1 ) {
	// If we only get a single key, auto-select it for convenience.
	newKeyId = keys[ 0 ];
}

if ( keys.includes( activePartnerKeyId ) ) {
	// If the active key id is for a valid key, select it.
	newKeyId = activePartnerKeyId;
}

if ( newKeyId ) {
	dispatch( setActivePartnerKey( newKeyId ) );
}
```
Also it will cover the situation where a cached key is no longer available and the user has only one available key now but they will still be asked to choose it because of this early return:
```
if ( ! isPartnerKeyAvailable ) {
	// do something else
	return;
}
```